src/nat/observability/exporter/processing_exporter.py

```
@@ -304,43 +304,29 @@ def _validate_insertion_compatibility(self, processor: Processor, position: int)
-            logger.warning(
-                "Cannot use issubclass() for type compatibility check between "
-                "%s (%s) and %s (%s). Skipping compatibility check.",
-                source_processor.__class__.__name__,
-                source_type,
-                target_processor.__class__.__name__,
-                target_type)
+        # Use Pydantic-based type compatibility checking
+        if not source_processor.is_output_compatible_with(target_processor.input_type):
+            raise ValueError(f"Processor {target_processor.__class__.__name__} input type {target_type} "
+                             f"is not compatible with {relationship} {source_processor.__class__.__name__} "
+                             f"output type {source_type}")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)